### PR TITLE
feat!: Voice region strings and VoiceRegion data class

### PR DESCRIPTION
**Pull request by onerandomusername**
## Summary

removes the disnake.VoiceRegion enum and replaces it with a custom class
## Checklist

- [x] If code changes were made, then they have been tested
    - [x] I have updated the documentation to reflect the changes
    - [x] I have formatted the code properly by running `task lint`
    - [x] I have type-checked the code by running `task pyright`
- [ ] This PR fixes an issue
- [x] This PR adds something new (e.g. new method or parameters)
- [x] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: disnake/channel.py [hunk 1]
```diff
@@ -1249,12 +1241,15 @@ async def edit(self, *, reason=None, **options):
         overwrites: :class:`Mapping`
             A :class:`Mapping` of target (either a role or a member) to
             :class:`PermissionOverwrite` to apply to the channel.
-        rtc_region: Optional[:class:`VoiceRegion`]
+        rtc_region: Optional[Union[:class:`str, `:class:`VoiceRegion`]]
```
[the remark this answers is not in the document]
Comment (onerandomusername) [reply disnake/channel.py]: https://github.com/DisnakeDev/disnake/pull/477/commits/c13977c353f84414c123f3bd9388d8720a92338c

File: disnake/client.py [hunk 2]
```diff
@@ -1860,6 +1865,33 @@ async def delete_invite(self, invite: Union[Invite, str]) -> None:
         invite_id = utils.resolve_invite(invite)
         await self.http.delete_invite(invite_id)
 
+    # Voice region stuff
+
+    async def fetch_voice_regions(self, guild_id: Optional[int] = None) -> List[VoiceRegion]:
+        """Retrieves a list of :class:`VoiceRegion`\\s.
```
[the remark this answers is not in the document]
Comment (onerandomusername) [reply disnake/client.py]: https://github.com/DisnakeDev/disnake/pull/477/commits/c13977c353f84414c123f3bd9388d8720a92338c

File: disnake/channel.py [hunk 3]
```diff
@@ -1249,12 +1241,15 @@ async def edit(self, *, reason=None, **options):
...
-        rtc_region: Optional[:class:`VoiceRegion`]
+        rtc_region: Optional[Union[:class:`str`, :class:`VoiceRegion`]]
             The new region for the voice channel's voice communication.
             A value of ``None`` indicates automatic voice region detection.
 
             .. versionadded:: 1.7
 
+            .. versionchanged:: 2.5
+                No longer a ``VoiceRegion`` instance.
+
```
Comment (shiftinv) [inline disnake/channel.py:1252]:
```suggestion
```

Comment (onerandomusername) [reply disnake/channel.py]: https://github.com/DisnakeDev/disnake/pull/477/commits/3b5c5648759b40ba97819270769aa5f493054f4a

File: disnake/channel.py [hunk 4]
```diff
@@ -1042,7 +1034,7 @@ class VoiceChannel(disnake.abc.Messageable, VocalGuildChannel):
         The channel's preferred audio bitrate in bits per second.
     user_limit: :class:`int`
         The channel's limit for number of members that can be in a voice channel.
-    rtc_region: Optional[:class:`VoiceRegion`]
+    rtc_region: Optional[:class:`str`]
         The region for the voice channel's voice communication.
         A value of ``None`` indicates automatic voice region detection.
```
[the remark this answers is not in the document]
Comment (onerandomusername) [reply disnake/channel.py:1040]: https://github.com/DisnakeDev/disnake/pull/477/commits/3b5c5648759b40ba97819270769aa5f493054f4a

File: disnake/guild.py [hunk 5]
```diff
@@ -1687,7 +1700,7 @@ async def edit(
         splash: Optional[AssetBytes] = MISSING,
         discovery_splash: Optional[AssetBytes] = MISSING,
         community: bool = MISSING,
-        region: Optional[Union[str, VoiceRegion]] = MISSING,
+        region: Optional[str] = MISSING,
```
Comment (shiftinv) [inline disnake/guild.py:1703]:
```suggestion
```

File: disnake/voice_region.py [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
+class VoiceRegion:
+    """Represents a Discord voice region.
+
+    .. versionadded:: 2.5
+
+    Attributes
+    ----------
+    id: :class:`str`
+        unique ID for the region.
```
Comment (shiftinv) [inline disnake/voice_region.py:43]:
```suggestion
        Unique ID for the region.
```

File: disnake/channel.py [hunk 7]
```diff
@@ -1565,9 +1560,13 @@ class StageChannel(VocalGuildChannel):
         The channel's preferred audio bitrate in bits per second.
     user_limit: :class:`int`
         The channel's limit for number of members that can be in a stage channel.
-    rtc_region: Optional[:class:`VoiceRegion`]
+    rtc_region: Optional[:class:`str`]
         The region for the stage channel's voice communication.
         A value of ``None`` indicates automatic voice region detection.
+
+        .. versionchanged:: 2.5
+                No longer a ``VoiceRegion`` instance.
```
Comment (shiftinv) [inline disnake/channel.py:1568]:
```suggestion
            No longer a ``VoiceRegion`` instance.
```

**Review by shiftinv — COMMENTED** [review]
mostly nitpicks

File: disnake/channel.py (hunk 7 above)
Comment (onerandomusername) [reply disnake/channel.py]: https://github.com/DisnakeDev/disnake/pull/477/commits/99daf1f396af005580f40774ab91d4fc1e74acac

File: disnake/voice_region.py (hunk 6 above)
Comment (onerandomusername) [reply disnake/voice_region.py]: https://github.com/DisnakeDev/disnake/pull/477/commits/99daf1f396af005580f40774ab91d4fc1e74acac

File: disnake/guild.py (hunk 5 above)
Comment (onerandomusername) [reply disnake/guild.py]: https://github.com/DisnakeDev/disnake/pull/477/commits/a8f60ba14822f7751972aeec4b00632ff22bc908

**Review by shiftinv — APPROVED** [review]
lgtm

**Pull request merged**